velox/connectors/hive/HiveConnector.h

```diff
@@ -92,38 +96,121 @@ class HiveTableHandle : public ConnectorTableHandle {
   const core::TypedExprPtr remainingFilter_;
 };
 
+/// Location related properties of the Hive table to be written
+class LocationHandle {
+ public:
+  enum class TableType {
+    kNew, // Write to a new table to be created.
+    kExisting, // Write to an existing table.
+    kTemporary, // Write to a temporary table.
+  };
+  enum class WriteMode {
+    // Write to a staging directory and then move to the target directory
+    // after write finishes.
+    kStageAndMoveToTargetDirectory,
+    // Directly write to the target directory to be created.
+    kDirectToTargetNewDirectory,
+    // Directly write to the existing target directory.
+    kDirectToTargetExistingDirectory,
+  };
+
+  LocationHandle(
+      const std::string& targetPath,
```

Comment:
Had missed this place. Revising.

Masha, just want to learn, why do we prefer passing by value plus std::move, compared with passing by reference and copy initialization? Because it allows callers to pass in rvalue?

Reply:
Passing by reference + copy always requires 1 copy. Passing by value + std::move sometimes doesn't require a copy, e.g. foo(std::move(value)) doesn't require a copy.